psi-java/src/main/kotlin/io/spine/tools/psi/java/PsiStatements.kt

```diff
@@ -34,34 +34,70 @@ import com.intellij.psi.PsiElementFactory
  * A list of statements extracted from the given [PsiCodeBlock],
  * excluding the surrounding braces.
  *
- * This type addresses several challenges:
+ * The presence of this type addresses several challenges:
  *
  * 1. In PSI, a code block cannot be created without curly braces. As a result, even if
  *    you only need a list of statements, you must create a full block with braces.
  * 2. Inserting such a block into another code block requires range copying
  *    to skip the braces.
  * 3. While a list of statements can be [retrieved][PsiCodeBlock.getStatements] directly from
- *    the code block, formatting elements like whitespace and newlines are omitted, leaving
- *    only the raw statements. Adding these statements individually to a method body often
- *    results in invalid or non-compilable Java code.
+ *    the code block, formatting elements like whitespace and newlines are omitted,
+ *    leaving only the raw statements. Adding these statements to a method body may
+ *    result in non-compilable Java code.
  */
 public class PsiStatements(codeBlock: PsiCodeBlock) {
 
     /**
-     * All children of [PsiCodeBlock] without right and left braces.
+     * The underlying [PsiCodeBlock], which actually manages children
+     * of this [PsiStatements].
+     *
+     * PSI elements are linked with each other, and they cannot be managed
+     * separately without their parents, children and siblings relations.
+     * As a result, instead of copying only the children of interest from
+     * the passed block, we have to copy the whole block using [PsiElement.copy].
+     * Having a hard copy of the passed block protects from the outside changes
+     * to the original block, and allows PSI to manage the elements correctly
+     * when modified by [append] and [prepend] method.
      */
-    private val children = codeBlock.children
-        .copyOfRange(1, codeBlock.children.size - 1)
+    private val delegate: PsiCodeBlock = codeBlock.copy() as PsiCodeBlock
 
     /**
      * Returns the first child of this element.
      */
-    public val firstChild: PsiElement = children.first()
+    public val firstChild: PsiElement
```

Comment:
Maybe it's `firstStatement` to avoid the confusion with the braces being also children?

Reply:
The first and last children may be formatting elements.